samples/drivers/flash_shell/src/main.c

```diff
@@ -336,7 +494,58 @@ static int cmd_write(const struct shell *shell, size_t argc, char **argv)
 		}
 	}
 
-	err = do_write(shell, offset, buf, i, true);
+	if (!unaligned) {
+		err = do_write(shell, offset, buf, i, true);
+	} else {
+		err = do_write_unaligned(shell, offset, buf, i, true);
+	}
+
+exit:
+	return err;
+}
+
+static int cmd_write(const struct shell *shell, size_t argc, char **argv)
+{
+	return cmd_write_template(shell, argc, argv, false);
+}
+
+static int cmd_write_unaligned(const struct shell *shell, size_t argc, char **argv)
+{
+	return cmd_write_template(shell, argc, argv, true);
+}
+
+static int cmd_write_pattern(const struct shell *shell, size_t argc, char **argv)
+{
+	int err = check_flash_device(shell);
+	unsigned long int offset, len, i;
+	static uint8_t *buf;
+
+	if (err) {
+		goto exit;
+	}
+
+	if (parse_ul(argv[1], &offset) || parse_ul(argv[2], &len)) {
+		PR_ERROR(shell, "Invalid arguments.\n");
+		err = -EINVAL;
+		goto exit;
+	}
+
+	buf = k_malloc(len);
+
+	if (!buf) {
+		PR_ERROR(shell, "No heap memory for data pattern\n");
+		err = -ENOMEM;
+		goto free_buffers;
```

Comment:
This error path only happens if buf is NULL, so there isn't much reason to go to a free_buffers label, exit would be fine.

Reply:
You're right, will remove label.